src/lithopscloud/modules/utils.py

```diff
@@ -107,63 +110,165 @@ def free_dialog(msg, default=None):
+
     questions = [
         inquirer.List('answer',
                       message=msg,
                       choices=choices,
                       default=default,
-                      )] if not multiple_choice else \
-        [inquirer.Checkbox('answer',
-                           message=msg,
-                           choices=choices,
-                           default=default,
-                           )]
+                      carousel=carousel
+                      )]
 
     answers = inquirer.prompt(questions, raise_keyboard_interrupt=True)
 
-    while not answers['answer'] and multiple_choice:
-        print("You must choose at least one option.\n"
-              "To pick an option please use the right arrow key '->' to select and the left arrow key '<-' to cancel.")
-        answers = inquirer.prompt(questions)
-
     return answers
 
-#ToDo add as decorator to code_engine's ce_client.get_kubeconfig
-def retry_on_except(func):
-    retries = 0  # constants will be replaced by arguments to decorator:
-    sleep_duration = 0
-    def decorated_func(*args, **kwargs):
-        name = func.__name__
-        e = None
-        for retry in range(retries):
-            try:
-                result = func(*args, **kwargs)
-                return result
-            except Exception as e:
-                msg = f"Error in {name}, {e}, retries left {retries - retry}"
-                print(msg)
-                print(e)
-
-                args[0]._init()
-                time.sleep(sleep_duration)
-
-        raise e
-
-    return decorated_func
+
+def retry_on_except(retries, sleep_duration):
+    """A decorator that calls the decorated function up to a number of times equals to 'retires' with a given
+      'sleep_duration' in between"""
+
+    def retry_on_except_warpper(func):
+        def func_wrapper(*args, **kwargs):
+            function_name = func.__name__
+
+            for retry in range(retries):
+                try:
+                    result = func(*args, **kwargs)
+                    return result
+                except Exception as e:
+                    msg = f"Error in {function_name}, {e}, retries left {retries - retry - 1}"
+                    print(color_msg(msg,color=Color.RED))
+
+                    if retry < retries - 1:  # avoid sleeping after last failure
+                        time.sleep(sleep_duration)
+            print(color_msg(f"\nConfig tool was terminated, "
+                            f"as it can't progress without the successful activation of {function_name}",
+                            color=Color.RED))
+            sys.exit()
+
+        return func_wrapper
+
+    return retry_on_except_warpper
+
+
+def test_config_file(config_file_path):
+    """testing the created config file with a simple test  """
+
+    cmd = f"lithops test -c {config_file_path}"
+
+    virtual_env = get_confirmation("Is your lithops installed within a virtual environment?")['answer']
```

Comment:
I think we can assume that user uses same python setup to run lithops and lithopscloud